components/ApprovalWidget/ApproveDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react';
+import Dialog from 'components/Dialog/Dialog';
+import axios from 'axios';
+import { Submission } from 'data/flexibleForms/forms.types';
+import Banner from 'components/FlexibleForms/Banner';
+
+interface Props {
+  isOpen: boolean;
+  setOpen: (value: boolean) => void;
+  submission: Submission;
+}
+
+const ApproveDialog = ({
+  submission,
+  isOpen,
+  setOpen,
+}: Props): React.ReactElement => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [status, setStatus] = useState<string | false>(false);
+
+  const handleApprove = async () => {
+    try {
+      setLoading(true);
+      await axios.post(`/api/submissions/${submission.submissionId}/approvals`);
+      window.location.reload;
```

Comment:
Why?
Wouldn't it make more sense to SetLoading to false and to close the dialog?

Maybe a toast message to indicate success.

Reply:
we don't have toasts or anything in the ui yet. this is what i was meaning to follow up with in a later pr rather than making this one bigger